### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by slipros**

### CODE REVIEW

File: formatter/json.go [hunk 1]
```diff
@@ -16,8 +16,9 @@ func NewJsonPretty() *Json {
 	return &Json{pretty: true}
 }
 
-func (j *Json) Write(data any, w http.ResponseWriter) error {
+func (j *Json) Write(w http.ResponseWriter, statusCode int, data any) error {
 	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(statusCode)
```
Comment (raoptimus) [inline formatter/json.go:21]: it has in handler
```
 writer.Write(w, resp.StatusCode()
```
I think the place is not good

File: formatter/json.go [hunk 2]
```diff
@@ -16,8 +16,9 @@ func NewJsonPretty() *Json {
 	return &Json{pretty: true}
 }
 
-func (j *Json) Write(data any, w http.ResponseWriter) error {
+func (j *Json) Write(w http.ResponseWriter, statusCode int, data any) error {
```
Comment (raoptimus) [inline formatter/json.go:19]: w http.ResponseWriter don't need here

File: dependency.go [hunk 3]
```diff
@@ -1,29 +1,28 @@
...
-	}
-)
+type Factory interface {
+	FactoryAPI
+
+	FormatWriter() FormatWriter
+}
+
+type FactoryAPI interface {
+	CreateResponse(ctx context.Context, statusCode int, data any) *DataResponse
```
Comment (raoptimus) [inline dependency.go]: the function must have in Factory because Factory is using for html format

File: handler/data_response.go [hunk 4]
```diff
@@ -1,28 +0,0 @@
-package handler
```
Comment (raoptimus) [inline handler/data_response.go:1]: why did you delete this code?

**Review by raoptimus — CHANGES_REQUESTED** [review]

File: handler/handler.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+			}
+
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+// NewHTTPHandler - оборачивает http вызов в response.Handler и возвращает http.Handler.
+func NewHTTPHandler(f response.Factory, h response.Handler) http.Handler {
```
Comment (raoptimus) [inline handler/handler.go]: DataResponse

File: handler/handler.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package handler
+
+import (
+	"net/http"
+
+	response "github.com/raoptimus/data-response.go"
+)
+
+// NewHTTPHandlerFunc - оборачивает http вызов в response.Handler и возвращает http.HandlerFunc.
+func NewHTTPHandlerFunc(f response.Factory, h response.Handler) http.HandlerFunc {
```
Comment (raoptimus) [inline handler/handler.go]: DataResponseFunc

**Review by raoptimus — CHANGES_REQUESTED** [review]

File: handler/handler.go (hunk 6 above)
Comment (slipros) [reply handler/handler.go]: Done

File: handler/handler.go (hunk 5 above)
Comment (slipros) [reply handler/handler.go]: Done

File: dependency.go (hunk 3 above)
Comment (slipros) [reply dependency.go]: `Factory` has all methods from `FactoryAPI`

Comment (raoptimus) [reply dependency.go]: it is not right for html format. html must use template and only CreateResponse. example: function CreateUnprocessableEntityResponse don't need for rendering template

**Review by raoptimus — CHANGES_REQUESTED** [review]

File: dependency.go (hunk 3 above)
Comment (slipros) [reply dependency.go]: Done
